src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -325,16 +325,17 @@ boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
     for (String hostName : allHosts) {
       LOG.debug("checking host '{}' for pending compactions and other repairs (can repair?)"
                 + " Run id '{}'", hostName, segment.getRunId());
-      JmxProxy hostProxy = null;
+      Optional<JmxProxy> hostProxy = Optional.absent();
       try{
-        Optional<HostMetrics> hostMetrics;
+        Optional<HostMetrics> hostMetrics = Optional.absent();
         try{
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)